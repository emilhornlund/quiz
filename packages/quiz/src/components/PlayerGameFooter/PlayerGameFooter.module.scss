@import '../../styles/colors';
@import '../../styles/helpers';

.main {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  background-color: rgba(255, 255, 255, 0.5);

  @include mobile {
    border-radius: $layout-gutter * 0.5;
    padding: $layout-gutter * 0.5;
    width: 100%;
  }

  @include tablet {
    border-radius: $layout-gutter * 0.75;
    padding: $layout-gutter * 0.75;
    width: $tablet-width;
  }

  @include desktop {
    border-radius: $layout-gutter;
    padding: $layout-gutter;
    width: $desktop-width;
  }

  .questions {
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: center;
    flex: 1;
    font-weight: bold;

    @include mobile {
      @include fontSize(16px);
    }

    @include tablet {
      @include fontSize(20px);
    }

    @include desktop {
      @include fontSize(24px);
    }
  }

  .nickname {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    flex: 1;
    font-weight: bold;

    @include mobile {
      @include fontSize(16px);
    }

    @include tablet {
      @include fontSize(20px);
    }

    @include desktop {
      @include fontSize(24px);
    }
  }

  .score {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    align-items: center;
    flex: 1;

    @include mobile {
      @include fontSize(16px);
    }

    @include tablet {
      @include fontSize(20px);
    }

    @include desktop {
      @include fontSize(24px);
    }

    span {
      font-weight: bold;
      background-color: $white;
      color: $gray-4;

      @include mobile {
        padding: 0 $layout-gutter * 0.5;
        border-radius: pxToRem(8px);
      }

      @include tablet {
        padding: 0 $layout-gutter * 0.75;
        border-radius: pxToRem(10px);
      }

      @include desktop {
        padding: 0 $layout-gutter;
        border-radius: pxToRem(12px);
      }
    }
  }
}