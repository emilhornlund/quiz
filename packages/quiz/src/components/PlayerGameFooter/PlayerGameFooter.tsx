import React, { FC } from 'react'

import styles from './PlayerGameFooter.module.scss'

export interface PlayerGameFooterProps {
  currentQuestion: number
  totalQuestions: number
  nickname: string
  totalScore: number
}

const PlayerGameFooter: FC<PlayerGameFooterProps> = ({
  currentQuestion,
  totalQuestions,
  nickname,
  totalScore,
}) => (
  <div className={styles.main}>
    <div className={styles.questions}>
      {currentQuestion} / {totalQuestions}
    </div>
    <div className={styles.nickname}>{nickname}</div>
    <div className={styles.score}>
      <span>{totalScore}</span>
    </div>
  </div>
)

export default PlayerGameFooter
