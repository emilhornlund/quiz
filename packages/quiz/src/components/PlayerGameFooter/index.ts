export type { PlayerGameFooterProps } from './PlayerGameFooter'
export { default } from './PlayerGameFooter'
