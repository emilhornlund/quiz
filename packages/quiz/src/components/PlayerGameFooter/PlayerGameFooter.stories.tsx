import type { Meta, StoryObj } from '@storybook/react'

import PlayerGameFooter from './PlayerGameFooter'

const meta = {
  component: PlayerGameFooter,
} satisfies Meta<typeof PlayerGameFooter>

export default meta
type Story = StoryObj<typeof meta>

export const Default = {
  args: {
    currentQuestion: 1,
    totalQuestions: 20,
    nickname: 'FrostyBear',
    totalScore: 10361,
  },
} satisfies Story
