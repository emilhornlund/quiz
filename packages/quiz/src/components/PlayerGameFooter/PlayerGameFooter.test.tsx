import { render } from '@testing-library/react'
import React from 'react'
import { describe, expect, it } from 'vitest'

import PlayerGameFooter from './PlayerGameFooter'

describe('PlayerGameFooter', () => {
  it('should render PlayerGameFooter with default props', () => {
    const { container } = render(
      <PlayerGameFooter nickname="FrostyBear" totalScore={10361} />,
    )

    expect(container).toMatchSnapshot()
  })
})
