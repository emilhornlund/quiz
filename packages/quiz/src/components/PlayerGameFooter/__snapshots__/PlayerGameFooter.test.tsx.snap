// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayerGameFooter > should render PlayerGameFooter with default props 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="questions"
    >
       / 
    </div>
    <div
      class="nickname"
    >
      FrostyBear
    </div>
    <div
      class="score"
    >
      <span>
        10361
      </span>
    </div>
  </div>
</div>
`;
